Java/core/src/main/java/com/amazon/randomcutforest/returntypes/SampleSummary.java

```diff
@@ -83,4 +92,62 @@ public SampleSummary(double[] point) {
         this.deviation = new float[point.length];
     }
 
+    public SampleSummary(int dimension, float[][] typicalPoints, float[] relativeLikelihood) {
+        this.addTypical(dimension, typicalPoints, relativeLikelihood);
+    }
+
+    public void addTypical(int dimension, float[][] typicalPoints, float[] relativeLikelihood) {
+        checkArgument(typicalPoints.length == relativeLikelihood.length, "incorrect lengths of fields");
+        this.summaryPoints = new float[typicalPoints.length][];
```

Comment:
This method completely overwrites the contents of the class. Is that the intention? Should this just be part of the constructor code?

Reply:
That was the intention -- the summary has both the basic statistical component and this "extra". We need to separate them -- we can have two constructors, but then repeating the basic statistical estimation would be too much. Let me take another shot.